### PR TITLE
fix: broken deploy

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- We should fix the deploy script on github actions as it had a typo
- We should fix the docker image build, it was using the `node_modules` folder from the host instead of the locally installed
- We should update nodejs to v20

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: packages/daemon/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+dist/
+node_modules/
+__tests__/
+node_modules/
```
Comment (luislhl) [inline packages/daemon/.dockerignore]: I believe this has no effect, but this one is duplicated.

File: packages/daemon/Dockerfile [hunk 2]
```diff
@@ -15,22 +15,29 @@ COPY package.json ./
...
 RUN which python
 
-RUN yarn set version berry
+RUN yarn set version 3.4.1
 
 RUN yarn install
 
 COPY . ./
 
+RUN ls -la /app
```
Comment (luislhl) [inline packages/daemon/Dockerfile]: Should we keep this?

File: packages/daemon/.dockerignore (hunk 1 above)
Comment (andreabadesso) [reply packages/daemon/.dockerignore]: Fixed, thanks!

File: packages/daemon/Dockerfile (hunk 2 above)
Comment (andreabadesso) [reply packages/daemon/Dockerfile]: We shouldn't, fixed!

**Review by luislhl — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Hathor Labs
```
Comment (luislhl) [inline Dockerfile]: Since you're adding this, should we use 2024?

File: packages/daemon/.dockerignore
Comment (luislhl) [inline packages/daemon/.dockerignore:1]: Is this file still needed, since there is one in the root folder, and the docker build is made there now?

**Review by luislhl — APPROVED** [review]

File: packages/daemon/.dockerignore
Comment (andreabadesso) [reply packages/daemon/.dockerignore:1]: It is not! Removed

File: Dockerfile (hunk 3 above)
Comment (andreabadesso) [reply Dockerfile]: Sure! Updated

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,4 @@ packages/wallet-service/.serverless
 packages/wallet-service/.webpack
 packages/wallet-service/.env*
 .yarn/
+.git
```
Comment (tuliomir) [inline .gitignore]: Question: I'm unsure if this has the expected behavior. The `.git` folder is always ignored by default, so why should we add it here?

If we want to ignore nested `.git` folders, will this command work? Or should we use `**/.git`?

Comment (andreabadesso) [reply .gitignore]: This wasn't supposed to be there, fixed!

Thanks

**Review by tuliomir — APPROVED** [review]
